test/linter-test.js

```diff
@@ -42,59 +42,27 @@ describe('linter', () => {
 
     it('uses new instance for different directory', async () => {
       const cwd2 = path.join(cwd, 'test');
-      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', 0, () => {});
-      await linter.invoke(cwd2, ['--stdin'], '\'use strict\';', 0, () => {});
+      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', 'a', () => {});
+      await linter.invoke(cwd2, ['--stdin'], '\'use strict\';', 'a', () => {});
 
       assert.equals(linter.cache.length, 2);
       assert.callCount(resolver.resolve, 4);
       assert.calledWithMatch(resolver.resolve, 'eslint', { paths: [cwd] });
       assert.calledWithMatch(resolver.resolve, 'eslint', { paths: [cwd2] });
     });
 
-    it('creates new instance if mtime is larger than first call', async () => {
-      const now = Date.now();
-      const clock = sinon.useFakeTimers(now);
-      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', now - 1000,
-        () => {});
+    it('creates new instance if hash differs from first call', async () => {
+      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', 'a', () => {});
       const cache1 = linter.cache.get(cwd);
 
-      clock.tick(1000);
-      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', now + 500,
-        () => {});
+      await linter.invoke(cwd, ['--stdin'], '\'use strict\';', 'b', () => {});
       const cache2 = linter.cache.get(cwd);
 
       assert.equals(linter.cache.length, 1);
       refute.same(cache1, cache2);
       refute.same(cache1.eslint, cache2.eslint, 'require.cache cleared');
       assert.callCount(resolver.resolve, 4);
     });
-
-    it('does not create new instance if mtime is lower than last call',
-      async () => {
```

Comment:
Removed because comparing equality of strings is a simpler case, which I think is covered well enough in the tests `reuses instance from cache` and `creates new instance if hash differs from first call`